### PR TITLE
AP_Notify: Correct the display area clear of the text message.

### PR DESCRIPTION
**Pull request by muramura**
There are times when you write 18 or more characters in the text message area. The text message area is cleared with space characters for 17 characters.
Therefore, the 18 th character, several 19 dots are displayed.
17308882_1416907415027175_6669827233328936807_n

Therefore, let the text message area be 20 bytes "19 characters + NULL character".
Let the message area be 19 bytes "18 characters + NULL character".

17353357_1416907698360480_2160631333533983551_n

Other
Since message buffer and text buffer are not initialized, characters are displayed until the message is received. Therefore, clear the message buffer and text buffer to 0.

17309276_1416932608357989_64033971302670459_n

### CODE REVIEW

File: libraries/AP_Notify/Display.cpp [hunk 1]
```diff
@@ -424,7 +424,7 @@ void Display::update_prearm(uint8_t r)
 void Display::update_gps(uint8_t r)
 {
 	static const char * gpsfixname[] = {"Other", "NoGPS","NoFix","2D","3D","DGPS", "RTK f", "RTK F"};
-	char msg [DISPLAY_MESSAGE_SIZE];
+	char msg [DISPLAY_MESSAGE_SIZE - 1];
```
Comment (magicrub) [inline libraries/AP_Notify/Display.cpp]: why -1 when the snprintf() uses the full length?

Comment (muramura) [inline libraries/AP_Notify/Display.cpp]: It is a specification that the message does not scroll.
In this case, up to 18 characters can be displayed.
Therefore, 18 characters plus NULL character is 19 bytes.

**Comment by rmackay9** [comment]
aha!  I've noticed this little issue.  nice to have a fix.

**Comment by muramura** [comment]
@OXINARF 
I will change about comments.
· Line 513: Change from "sizeof (msg)" to "sizeof (msg) - 1".
· Line 495: Change from "0" to "4".

**Comment by muramura** [comment]
@OXINARF 
Conflict changed.

**Comment by OXINARF** [comment]
@muramura Please check if https://github.com/OXINARF/ardupilot/tree/review/5880 fixes the problem. Thanks.

**Comment by muramura** [comment]
@OXINARF Thank you for improvement.
I replaced the following Display.cpp, Display.h with the latest source "APM: Copter V3.5-dev (be04cf12)" and confirmed it.
There is no dot display at the right end of the text message area.
All are clear.
https://github.com/OXINARF/ardupilot/tree/review/5880 

17425015_1422503297800920_764816413068303831_n

**Comment by OXINARF** [comment]
Ok, merged! Thanks @muramura!

**Comment by rmackay9** [comment]
I just tested this (from master) and it's working well, thanks!

**Pull request closed without merging**